src/babashka/curl.clj

```diff
@@ -145,34 +146,45 @@
       (.reset rdr))
     eof?))
 
+(defn- read-line [^java.io.InputStream is]
+  (let [sb (StringBuilder.)
+        res (loop [in is]
+              (let [c1 (.read is)]
+                (case c1
+                  ;; 10 = \n, 13 = \r
+                  (-1 10) (str sb)
+                  13 (let [c2 (.read is)]
+                        (when (and (not= c2 10) (not= c2 -1))
+                          (let [in (PushbackInputStream. is)]
```

Comment:
@rymndhng I'm not entirely sure about this part. This probably doesn't have an effect on the original input-stream? Should we write a test for it, which covers this part?

Reply:
Agreed a test will help here to check input streams with various line endings. I also believe you're correct that this won't have an effect on the `input-stream` which will skip characters if the next invocation of `read-line` handles text like this. 

One potential suggestion I have to get around this issue is to move `(PushbackInputStream. is)` into `curl-response->map` and change the typehint of `read-line` to be `^java.io.PushbackInputStream.`. This way we do not need to create a `PushbackInputStream` on-demand.